### ISSUE TITLE
Cannot return non-dictlike object from handler if BASE_RESPONSE_SCHEMA is set

### ISSUE THREAD

**Issue by lorenyu** [apiflask/apiflask#384]
If BASE_RESPONSE_SCHEMA is set and a non-dict like object is returned from a route, an exception `TypeError: argument of type '<type of object>' is not iterable` is thrown from [scaffold.py](https://github.com/apiflask/apiflask/blob/71c821ae59494b455d95f6df2e0f85c086652c23/src/apiflask/scaffold.py#L469) when checking `if data_key not in obj` (the `in` keyword doesn't work on all objects).

Environment:

- Python version:
- Flask version:
- APIFlask version:

**Comment by greyli** [comment]
Could you provide a minimal example (with the base schema and the view function)? By my understanding, the return value should be a dict when using the base schema.

**Comment by lorenyu** [comment]
Sounds good will do. I’m on vacation now but will aim to give an example once I’m back in the office

**Comment by lorenyu** [comment]
Sorry for the late reply! Here's a minimal example:

```python
from dataclasses import dataclass

from apiflask import APIFlask, Schema
from apiflask.fields import Field, String

app = APIFlask(__name__)


class BaseResponse(Schema):
    data = Field()
    message = String()


app.config["BASE_RESPONSE_SCHEMA"] = BaseResponse


class PetOut(Schema):
    name = String()


@dataclass
class Pet:
    name: str


@dataclass
class Response:
    data: Pet
    message: str


@app.get("/dict")
@app.output(PetOut)
def get_dict():
    data = {"name": "Kitty"}
    return {"message": "Success!", "data": data}


@app.get("/obj")
@app.output(PetOut)
def get_obj():
    data = Pet(name="Kitty")
    return Response(message="Success!", data=data)l
```

hitting `/dict` works, but `/obj` returns an Internal server error:

```
<img width="957" alt="image" src="https://user-images.githubusercontent.com/447859/213328647-cdb5182e-4f64-494f-99df-85ca2df54ff5.png">

```

**Comment by greyli** [comment]
Hi, this issue is fixed in [1.2.2](https://github.com/apiflask/apiflask/releases/tag/1.2.2). By the way, I'm curious why you want to return a data class instead of a dictionary for the base response. And I notice the `PetOut` schema is basically duplicated with `Pet`.
